aptos-move/framework/aptos-framework/sources/AuctionHouse.move

```diff
@@ -0,0 +1,209 @@
+        let auction_items = &mut auction_data.auction_items;
+        let auction_item = Table::borrow_mut(auction_items, token_id);
+        assert!(is_auction_complete(auction_item.start_time, auction_item.duration), ERROR_AUCTION_NOT_COMPLETE);
+
+        assert!(sender_addr == auction_item.current_bidder, ERROR_NOT_CLAIMABLE);
+
+        Event::emit_event<ClaimTokenEvent>(
+            &mut auction_data.claim_token_events,
+            ClaimTokenEvent { id: token_id },
+        );
+
+        let token = Option::extract(&mut auction_item.locked_token);
+        Token::deposit_token(sender, token);
+
+        // the auction item can be removed from the auctiondata of the seller once the token and coins are claimed
+        let locked_coins = &mut borrow_global_mut<CoinEscrow>(sender_addr).locked_coins;
+        // deposit the locked coins to the seller's sender if they have not claimed yet
+        if (Table::contains(locked_coins, token_id)){
+            Event::emit_event<ClaimCoinsEvent>(
+                &mut auction_data.claim_coins_events,
+                ClaimCoinsEvent { id: token_id },
+            );
+            let coins = Table::remove(locked_coins, token_id);
+            Coin::deposit<TestCoin>(seller, coins);
+        };
+        let AuctionItem{min_selling_price: _, duration: _, start_time: _, current_bid: _, current_bidder: _, locked_token: locked_token} = Table::remove(auction_items, token_id);
+        Option::destroy_none(locked_token);
+    }
+
+    public(script) fun claim_coins(sender: &signer, creator: address, collection_name: vector<u8>, name: vector<u8>) acquires CoinEscrow, AuctionData {
+        let token_id = Token::create_token_id_raw(creator, collection_name, name);
+        let sender_addr = Signer::address_of(sender);
+
+        let auction_data = borrow_global_mut<AuctionData>(sender_addr);
+        let auction_items = &mut auction_data.auction_items;
+
+        assert!(Table::contains(auction_items, token_id), ERROR_ALREADY_CLAIMED);
+        let auction_item = Table::borrow(auction_items, token_id);
+        assert!(is_auction_complete(auction_item.start_time, auction_item.duration), ERROR_AUCTION_NOT_COMPLETE);
+        assert!(sender_addr!=auction_item.current_bidder, ERROR_NOT_CLAIMABLE);
+
+        let locked_coins = &mut borrow_global_mut<CoinEscrow>(auction_item.current_bidder).locked_coins;
+        assert!(Table::contains(locked_coins, token_id), ERROR_ALREADY_CLAIMED);
+        Event::emit_event<ClaimCoinsEvent>(
+            &mut auction_data.claim_coins_events,
+            ClaimCoinsEvent { id: token_id },
+        );
+        let coins = Table::remove(locked_coins, token_id);
+        Coin::deposit<TestCoin>(sender_addr, coins);
+
+        // if the locked token from sender_addr doesn't contain the key token_id, the highest bidder has already claimed the token
+        // and the auction item can be removed from the auctiondata of the seller
```

Comment:
there is no check here to see if the auction item has been claimed... this destroy_none would abort if coin claim happened first afaiu.